FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_IPv6_Demo/IPv6_Multi_WinSim_demo/TCPEchoClient_SingleTasks.c

```diff
@@ -151,6 +154,68 @@
 
         return xReturn;
     }
+/*-----------------------------------------------------------*/
+
+    static BaseType_t vTCPCreateAndSendData( Socket_t xSocket,
+                                             char * pcTransmitBuffer,
+                                             volatile uint32_t * pulTxCount,
+                                             uint32_t ulBufferLength )
+    {
+        BaseType_t lStringLength;
+        BaseType_t lTransmitted;
+        char * pcDstBuffer;
+
+        #if USE_TCP_ZERO_COPY
+            BaseType_t xStreamBufferLength;
+            char * pcBuffer;
+        #endif /* USE_TCP_ZERO_COPY */
+
+        pcDstBuffer = pcTransmitBuffer;
+
+        #if USE_TCP_ZERO_COPY
+            /* Get a direct pointer to the TX head of the circular transmit buffer */
+            pcBuffer = ( char * ) FreeRTOS_get_tx_head( xSocket, &xStreamBufferLength );
+
+            /* Check if sufficient space is there in the stream buffer. If there
+             * isn't enough size use application provided buffer */
+            if( xStreamBufferLength >= ulBufferLength )
+            {
+                /* Use the TCP stream buffer to directly create the TX data */
+                pcDstBuffer = pcBuffer;
+            }
+        #endif /* USE_TCP_ZERO_COPY */
+
+        lStringLength = prvCreateTxData( pcDstBuffer, ulBufferLength );
+
+        /* Add in some unique text at the front of the string. */
+        sprintf( pcDstBuffer, "TxRx message number %u", ( unsigned ) *pulTxCount );
+        ( *pulTxCount )++;
+
+        #if USE_TCP_ZERO_COPY
+            /* Check to see if zero copy is used */
+            if( pcDstBuffer != pcTransmitBuffer )
+            {
+                /* Save a copy of the data in pcTransmitBuffer
+                 * which can be used to verify the echoed back data */
+                memcpy( pcTransmitBuffer, pcDstBuffer, lStringLength );
+
+                /* Set the buffer pointer as NULL to let the stack know
+                 * that its a zero copy */
+                pcDstBuffer = NULL;
+            }
+        #endif /* USE_TCP_ZERO_COPY */
+
+        /* Send the string to the socket. */
+        lTransmitted = FreeRTOS_send( xSocket,                /* The socket being sent to. */
+                                      ( void * ) pcDstBuffer, /* The data being sent. */
+                                      lStringLength,          /* The length of the data being sent. */
+                                      0 );                    /* No flags. */
```

Comment:
Do we need the ZERO_COPY flag here?

Reply:
For TCP zero copy, `FreeRTOS_send` identifies its a zero copy by checking if the `pcDstBuffer` is NULL, hence there is no zero copy flag for `FreeRTOS_send`.

Reply:
Got it Thanks